### PR TITLE
Revert extension handling changes

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

revert back to v4.0.4 

### _How did you test your changes?_

previous unit tests have been revived.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/CoinbaseWalletSDK.ts [hunk 1]
```diff
@@ -29,16 +25,8 @@ export class CoinbaseWalletSDK {
...
-      const ethereum = window.ethereum ?? window.top?.ethereum;
-      if (ethereum?.isCoinbaseBrowser) {
-        return ethereum as ProviderInterface;
-      }
-    } catch {
-      // Ignore
-    }
-    return new CoinbaseWalletProvider({ metadata: this.metadata, preference });
+    const params = { metadata: this.metadata, preference };
+    return getCoinbaseInjectedProvider(params) ?? new CoinbaseWalletProvider(params);
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletSDK.ts:29]: https://github.com/coinbase/coinbase-wallet-sdk/compare/v4.0.4...master#diff-39f5c3cd1d01da4693d4d2bffc017f605a8df261c485fb878c5ae723b0779a5a
image

File: packages/wallet-sdk/src/CoinbaseWalletSDK.test.ts [hunk 2]
```diff
@@ -2,28 +2,28 @@ import { CoinbaseWalletProvider } from './CoinbaseWalletProvider';
 import { CoinbaseWalletSDK } from './CoinbaseWalletSDK';
 import { ProviderInterface } from ':core/provider/interface';
 import { getFavicon } from ':core/type/util';
-
-const window = globalThis as { ethereum?: unknown };
+import { getCoinbaseInjectedProvider } from ':util/provider';
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletSDK.test.ts:5]: https://github.com/coinbase/coinbase-wallet-sdk/compare/v4.0.4...master#diff-5cf06b25464cf05fc95709cb941c7da0c72848654e4cf5e685317e3439b33ab3

File: packages/wallet-sdk/src/util/provider.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { LIB_VERSION } from '../version';
 import { standardErrors } from ':core/error';
-import { RequestArguments } from ':core/provider/interface';
+import { ConstructorOptions, ProviderInterface, RequestArguments } from ':core/provider/interface';
```
Comment (bangtoven) [inline packages/wallet-sdk/src/util/provider.ts:3]: https://github.com/coinbase/coinbase-wallet-sdk/compare/v4.0.4...master#diff-5e5fe1cbf8237e7fc878da6ee1e6f611e4ffbc47c991358b93f41ce788ead828

image

**Review by nateReiners — APPROVED** [review]

**Pull request merged**